### PR TITLE
fix: corrige crash recovery mobile + adiciona forgot password na web

### PR DESCRIPTION
**Pull request by coelhotv**
## Problema

Três issues identificadas após implementação do fluxo de sign-up (EXEC-SPEC-APP-SIGNUP.md):

1. **Crash mobile (TestFlight v0.3.3)** — app fechava ao clicar no link de recuperação de senha
2. **UI de reset não carregava no mobile** — mesmo sem crash, ResetPasswordScreen não renderizava
3. **Web/PWA não tinha fluxo de recuperação** — Auth.jsx tinha login/signup mas sem "Esqueci minha senha"

## Root Cause

`Navigation.jsx` chamava `supabase.auth.setSession()` sem `await` e sem `try/catch`. Promise rejeitada propagava pelo TurboModule bridge do React Native → `EXC_CRASH (SIGABRT)` em `ObjCTurboModule::performVoidMethodInvocation` (AP-138).

### Mobile — `Navigation.jsx`
- Handler `handleDeepLink` tornou-se async com `await` + `try/catch` em todas as calls
- Suporte adicionado para **PKCE flow** (`?code=` → `exchangeCodeForSession`) além do **implicit flow** (`#access_token=` → `setSession`)
- `queryString` agora isolada com `.split('#')[0]` para remover fragmento da URL antes do parsing
- `.getInitialURL().then()` com `.catch()` para cobrir rejeições do getInitialURL

### Web — `supabase.js` + `Auth.jsx`
- `sendPasswordReset(email)` exportado com `redirectTo: window.location.origin`
- `Auth.jsx`: novo modo forgot password via `ForgotPasswordCard` subcomponent
- Link "Esqueci minha senha" no modo login abaixo do campo senha
- Email preservado ao navegar entre modos
- Compatível com `ResetPasswordView.jsx` existente (via `detectSessionInUrl: true`)

## Edge Case Documentado

Usuário já logado que clica link de recovery tem a sessão substituída pelo Supabase — comportamento padrão deliberado sem código especial (R-224).

## Test Plan

- [ ] Abrir link de recovery no iOS (device físico) — sem crash, ResetPasswordScreen renderiza
- [ ] Web: Auth modal → "Esqueci minha senha" → form email → sucesso "Verifique seu email"
- [ ] Web: clicar link do email no browser → ResetPasswordView carrega
- [ ] `rtk npm run validate:agent` — 522/522 testes passando
- [ ] Lint: 0 erros

## AI Review Response

| Sugestão | Prioridade | Decisão | Motivo |
|----------|-----------|---------|-------|
| queryString parsing isola fragmento `#` antes do parse PKCE | High | Aplicado | Commit 24a4d839 |
| Inline style → classe CSS `.auth-forgot-password-wrapper` | Medium | Aplicado | Commit 2d387ee2 |

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/mobile/src/navigation/Navigation.jsx [hunk 1]
```diff
@@ -78,19 +78,43 @@ export default function Navigation() {
   }, [])
 
   useEffect(() => {
-    function handleDeepLink({ url }) {
+    async function handleDeepLink({ url }) {
       if (!url) return
+
+      // PKCE flow: dosiq://auth/callback?code=xxxxx
+      const queryString = url.split('?')[1]
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply apps/mobile/src/navigation/Navigation.jsx]: @gemini-code-assist fix applied in commit 24a4d839 — can you validate?

File: apps/web/src/views/Auth.jsx [hunk 2]
```diff
@@ -88,6 +175,18 @@ export default function Auth({ onAuthSuccess, onClose }) {
             />
           </div>
 
+          {isLogin && (
+            <div style={{ textAlign: 'right', marginTop: '-8px' }}>
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply apps/web/src/views/Auth.jsx]: @gemini-code-assist fix applied in commit 2d387ee2 — inline style movido para classe .auth-forgot-password-wrapper em Auth.css. Can you validate?
